### PR TITLE
Modernize module

### PR DESCRIPTION
**Pull request by ghoneycutt**
* Use data types for parameters
* Support modern versions of Puppet
* Support modern dependency versions
* Add REFERENCE.md

### CODE REVIEW

**Comment by ghoneycutt** [comment]
This pull request modernizes the module and after merging the open PR #64 can be closed as well as the open issues #60 and #67.

File: REFERENCE.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  log_path_template => 'C:\Logs\%%NAME%%.evtx'
+}
+
+#### Examples
+
+##### Manage the size of the Application log:
+
+```puppet
+
+```
```
Comment (smortex) [inline REFERENCE.md:33]: Seems to not catch example :-(

File: REFERENCE.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+##### Manage several custom event logs under C:\Logs:
+
+```puppet
+
+```
```
Comment (smortex) [inline REFERENCE.md:39]: Same

**Review by smortex — COMMENTED** [review]
Examples seems not to be processed as expected.

File: REFERENCE.md (hunk 1 above)
Comment (ghoneycutt) [reply REFERENCE.md]: Fixed

**Review by smortex — APPROVED** [review]
LGTM

**Comment by ghoneycutt** [comment]
@smortex Could you please merge it

**Pull request merged**